### PR TITLE
CONSOLE-3676: Convert app.jsx from class component to functional comp…

### PR DESCRIPTION
**Pull request by tvu20**
…onent

app.jsx converted from a class component to a functional component in order to use react-router v6 hooks

### CODE REVIEW

File: frontend/public/components/app.jsx [hunk 1]
```diff
@@ -80,37 +80,57 @@ const EnhancedProvider = ({ provider: ContextProvider, useValueHook, children })
...
-    this.previousMobileState = this._isMobile();
-    this.previousDrawerInlineState = this._isLargeLayout();
-
-    this.state = {
-      isMastheadStacked: this._isMobile(),
-      isNavOpen: this._isDesktop(),
-      isDrawerInline: this._isLargeLayout(),
-    };
-  }
+const App_ = (props) => {
```
Comment (rhamilto) [inline frontend/public/components/app.jsx:83]: Seems like a good opportunity to rename?

```suggestion
const App = (props) => {
```

**Review by rhamilto — COMMENTED** [review]
Awesome job, @tvu20!  One nit.  @jhadvig, thoughts on renaming?

**Review by rhamilto — APPROVED** [review]
/lgtm

**Comment by rhamilto** [comment]
@tvu20, it looks like there are some failing tests in need of investigation.

File: frontend/public/components/app.jsx (hunk 1 above)
Comment (jhadvig) [reply frontend/public/components/app.jsx]: 👍 on renaming

**Review by jhadvig — APPROVED** [review]
/lgtm

**Comment by tvu20** [comment]
This is technical debt cleanup and has no user-facing impact, so I am manually assigning docs and px approvals.

/assign @XiyunZhao for QE approval.
/label docs-approved
/label px-approved

**Comment by yanpzhan** [comment]
Had a regression test on the cluster launched against the pr, there is no regression issue found.
/label qe-approved

**Pull request merged**